src/main/java/com/google/sps/DataGraph.java

```diff
@@ -327,4 +329,69 @@ private void dfsVisit(GraphNode gn, Map<GraphNode, Boolean> visited, int depthRe
       }
     }
   }
+
+  /**
+   * Returns a MutableGraph with nodes that are at most a certain radius from a given node If the
+   * radius is less than 0 or the node specified isn't present, return an emptu graph Since maxDepth
+   * was implemented as DFS, we use BFS for *diversity*
+   *
+   * @param name the name of the node to search for
+   * @param radius the distance from the node to search for parents and children
+   * @return a graph comprised of only nodes and edges within a certain distance from the specified
+   *     node
+   */
+  public MutableGraph<GraphNode> getReachableNodes(String name, int radius) {
+    if (radius < 0) {
+      return GraphBuilder.directed().build(); // If max depth below 0, then return an emtpy graph
+    }
+
+    HashMap<String, GraphNode> graphNodesMap = this.graphNodesMap();
+
+    if (!graphNodesMap.containsKey(name)) {
+      return GraphBuilder.directed()
+          .build(); // If the specified node is not found, return an empty graph
+    }
+
+    MutableGraph<GraphNode> graph = this.graph();
+    GraphNode tgtNode = graphNodesMap.get(name);
+
+    Map<GraphNode, Boolean> visited = new HashMap<>();
+
+    HashSet<GraphNode> nextLayer;
+    ArrayDeque<GraphNode> queue = new ArrayDeque<>();
+
+    queue.add(tgtNode); // Adds the searched node to the queue
+
+    for (int i = 0; i <= radius; i++) {
+      // Break out early if queue is empty
+      if (queue.size() == 0) {
+        break;
+      }
+      nextLayer = new HashSet<>();
+      while (!queue.isEmpty()) {
+        GraphNode curr = queue.poll();
+
+        if (!visited.containsKey(curr)) {
+          visited.put(curr, true);
+
+          // Adds the children
+          for (GraphNode child : graph.successors(curr)) {
+            if (!visited.containsKey(child)) {
+              nextLayer.add(child);
+            }
+          }
+          // Adds the parents
```

Comment:
I thought we decided in poker planning that we would only return all reachable children and ancestors up to the root. Here since the parents and children are processed in one loop, it seems like we would also add other children of ancestors.

Reply:
ohhh ok my bad! so if you have B<--A-->C and looked for b, then it would just be A (and not C)?

Reply:
ok changed!